osu.Game/Screens/Select/Carousel/CarouselGroup.cs

```diff
@@ -92,7 +93,7 @@ public override void Filter(FilterCriteria criteria)
                 if (comparison != 0)
                     return comparison;
 
-                return x.ItemID.CompareTo(y.ItemID);
+                return -x.CompareTo(new FilterCriteria { Sort = SortMode.DateAdded }, y);
```

Comment:
you cannot allocate an object inside the comparer for every comparison. people have tens of thousands of beatmaps in their databases. best-case scenario for a sort is O(n log n). this will obliterate performance.